libs/search-widget/src/viewer/viewer.store.ts

```diff
@@ -184,6 +211,25 @@ export function clearSearch() {
   viewerStore.onlySelected.next(false);
 }
 
+export function setParagraphLabels(labels: Classification[], paragraph: WidgetParagraph) {
+  const resource = viewerStore.resource.getValue();
+  const saving = viewerStore.savingLabels.getValue();
+  if (!resource || saving) return;
+  viewerStore.savingLabels.next(true);
+  const paragraphId = getParagraphId(resource.id, paragraph);
+  setLabels(resource, paragraph.fieldId, paragraph.fieldType.slice(0, -1), paragraphId, labels)
```

Comment:
you can get `fieldId` and `fieldType` from `viewerStore.currentField`

Reply:
Fixed